client/main.lua

```diff
@@ -1,44 +1,207 @@
-end
-
---- Create a blip for the property/apartment
----@param blipId integer | string property id
----@param name string name that will be displayed on the blip
----@param coords vector3
----@param blip? integer
----@param color? integer
----@param size? integer
-function AddBlip(blipId, name, coords, blip, color, size)
-    if not blipId or not coords or not blip or not color or not size or not name then return end
-    blips[blipId] = AddBlipForCoord(coords.x, coords.y, coords.z)
-    SetBlipSprite(blips[blipId], blip)
-    SetBlipDisplay(blips[blipId], 4)
-    SetBlipScale(blips[blipId], size)
-    SetBlipColour(blips[blipId], color)
-    SetBlipAsShortRange(blips[blipId], true)
-    BeginTextCommandSetBlipName("STRING")
-    AddTextComponentString(name)
-    EndTextCommandSetBlipName(blips[blipId])
-end
---#endregion Functions
-
----@param players table
----@param conceal boolean
-local function concealPlayers(players, conceal)
-    if GetInvokingResource() ~= nil then return end
-    if not players then return end
-
-    for _, player in ipairs(players) do
-        NetworkConcealPlayer(player, conceal, false)
-    end
-end
-
-RegisterNetEvent('qbx-property:client:concealPlayers', concealPlayers)
\ No newline at end of file
+local blips = {}
+InteriorZones = {}
+
+--#region Functions
+--- Create the zones inside the property
+---@param coords table
+---@param propertyId number | nil
+---@param isVisit boolean
+function CreatePropertyInteriorZones(coords, propertyId, isVisit)
+    if next(InteriorZones) then
+        for _, v in pairs(InteriorZones) do
+            v:remove()
+        end
+        InteriorZones = {}
+    end
+    local customZones = propertyId and lib.callback.await('qbx-properties:server:GetCustomZones', false, propertyId) or {}
+
+    InteriorZones.entrance = lib.points.new({
+        coords = customZones?.entrance?.xyz or coords.entrance.xyz,
+        distance = 7.5,
+    })
+
+    function InteriorZones.entrance:nearby()
+        if not self then return end
+        if not self.currentDistance then return end
+        local marker = Config.InteriorZones.entrance.marker
+        DrawMarker(marker.type,
+            self.coords.x, self.coords.y, self.coords.z + marker.offsetZ, -- coords
+            0.0, 0.0, 0.0, -- direction?
+            0.0, 0.0, 0.0, -- rotation
+            marker.scale.x, marker.scale.y, marker.scale.z, -- scale
+            marker.color.r, marker.color.g, marker.color.b, marker.color.a, -- color RBGA
+            false, false, 2, false, nil, nil, false
+        )
+
+        if self.currentDistance < 1 then
+            SetTextComponentFormat("STRING")
+            AddTextComponentString(Lang:t('interiorZones.leave'))
+            DisplayHelpTextFromStringLabel(0, 0, 1, 20000)
+            if IsControlJustPressed(0, 38) then
+                TriggerServerEvent('qbx-properties:server:leaveProperty', propertyId, cache.vehicle or false)
```

Comment:
The `or false` here is unnecessary as cache.vehicle is already false whenever it doesn't hold a vehicle